docs/src/guide/installation.md

```diff
@@ -313,6 +312,105 @@ avrdude --version
+
+Install Ubuntu-22.04 LTS as WSL2 distribution:
+See the [general WSL doc][wsl-install], but as of this writing
+`wsl --list --online` will not list Ubuntu 22.04.1 LTS, but it is
+[available in the Microsoft Store][windows-store-ubuntu-22-04-1-lts].
+Install it from there via mouse clicks.
+Make sure the WSL2 instance is running by opening a terminal via start menu.
+
+Physically attach the microcontroller board to a USB port and
+[bridge the USB hub to which the device is attached][wsl-connect-usb].
+Also [bridge the USB device to Linux][connect-usb] using [usbipd][usbipd]:
+
+```powershell
+PS C:\Windows\system32> usbipd wsl list
+BUSID  VID:PID    DEVICE                                STATE
+3-2    1bcf:0005  USB-Eingabegerät                      Not attached
+3-3    0483:374b  ST-Link Debug, USB-...                Attached - Ubuntu-22.04
+4-1    138a:003d  Synaptics FP Sensors (WBF) (PID=003d) Not attached
+4-3    04f2:b370  HP HD Webcam [Fixed]                  Not attached
+```
+
+Check the BUSID parameter and use the one that corresponds to your device in
+further commands. In this example it is the `3-3`.
+
+```powershell
+PS C:\Windows\system32> usbipd wsl attach --busid 3-3
+usbipd: info: Using default distribution 'Ubuntu-22.04'.
+```
+
+!!! tip "Unreliable connection"
+    If the connection is not reliable, it might be necessary to detach/attach
+    several times until the connection is established:
+
+    ```powershell
+    PS C:\Windows\system32> usbipd wsl detach --busid 3-3
+    PS C:\Windows\system32> usbipd wsl attach --busid 3-3
+    usbipd: info: Using default distribution 'Ubuntu-22.04'.
+    ```
+
+In WSL2 Ubuntu configure the USB ports as user accessible:
+Use an editor to create a new `udev` rules file with a high enough number in
+the filename: `sudo nano /etc/udev/rules.d/100-udev-stm.rules`.
+Add some rules similar to [the ones used in OpenOCD][udev-rules-openocd]:
+
+```
+# STMicroelectronics ST-LINK V1
+ATTRS{idVendor}=="0483", ATTRS{idProduct}=="3744", MODE="660", GROUP="plugdev", TAG+="uaccess"
+
+# STMicroelectronics ST-LINK/V2
+ATTRS{idVendor}=="0483", ATTRS{idProduct}=="3748", MODE="660", GROUP="plugdev", TAG+="uaccess"
+
+# STMicroelectronics ST-LINK/V2.1
+ATTRS{idVendor}=="0483", ATTRS{idProduct}=="374b", MODE="660", GROUP="plugdev", TAG+="uaccess"
+ATTRS{idVendor}=="0483", ATTRS{idProduct}=="3752", MODE="660", GROUP="plugdev", TAG+="uaccess"
+
+# STMicroelectronics STLINK-V3
+ATTRS{idVendor}=="0483", ATTRS{idProduct}=="374d", MODE="660", GROUP="plugdev", TAG+="uaccess"
+ATTRS{idVendor}=="0483", ATTRS{idProduct}=="374e", MODE="660", GROUP="plugdev", TAG+="uaccess"
+ATTRS{idVendor}=="0483", ATTRS{idProduct}=="374f", MODE="660", GROUP="plugdev", TAG+="uaccess"
+ATTRS{idVendor}=="0483", ATTRS{idProduct}=="3753", MODE="660", GROUP="plugdev", TAG+="uaccess"
+ATTRS{idVendor}=="0483", ATTRS{idProduct}=="3754", MODE="660", GROUP="plugdev", TAG+="uaccess"
+```
```

Comment:
I don't understand why this is necessary, I assume this step is not needed at all (a reboot should do).
Because the [Ubuntu OpenOCD package](https://packages.ubuntu.com/jammy/amd64/openocd/filelist) already installs [`/etc/udev/rules.d/60-openocd.rules`](https://github.com/openocd-org/openocd/blob/master/contrib/60-openocd.rules).

Reply:
If already, one should copy the whole file, so that not only ST-LINK debuggers work.

Reply:
I removed it.